src/common/utils/channel/enlist-request-channel-listener-injection-token.ts

```diff
@@ -4,11 +4,11 @@
  */
 import { getInjectionToken } from "@ogre-tools/injectable";
 import type { RequestChannel } from "./request-channel-injection-token";
-import type { RequestChannelListener } from "./request-channel-listener-injection-token";
+import type { RequestChannelHandlerDescriptor } from "./request-channel-listener-injection-token";
 
 export type EnlistRequestChannelListener = <
   TChannel extends RequestChannel<any, any>,
->(listener: RequestChannelListener<TChannel>) => () => void;
+>(listener: RequestChannelHandlerDescriptor<TChannel>) => () => void;
```

Comment:
Nit: `descriptor` is a loaded term which already has a purpose in JavaScript, making this a tad confusing.